migration/migration-table.sql

```diff
@@ -0,0 +1,49 @@
+-- migration-table.sql
+DO
+$migration_table$
+    DECLARE
+        _current_user_id oid = NULL;
+        _ps_catalog_schema_owner_id oid = NULL;
+        _migration_table_owner_id text = NULL;
+    BEGIN
+        SELECT pg_user.usesysid
+        INTO STRICT _current_user_id
+        FROM pg_catalog.pg_user
+        WHERE pg_user.usename = current_user;
+
+        SELECT pg_namespace.nspowner
+        INTO _ps_catalog_schema_owner_id
+        FROM pg_catalog.pg_namespace
+        WHERE pg_namespace.nspname = '_ps_catalog';
+
+        IF _ps_catalog_schema_owner_id IS NOT NULL THEN
+            IF _ps_catalog_schema_owner_id != _current_user_id  THEN
+                RAISE 'Schema "_ps_catalog" is owned by the wrong user';
```

Comment:
Here and below would be helpful to state what is considered "the right user".

Reply:
I've addressed this. Although in writing the comment I realised that I'm not sure that this method is entirely correct: especially with the non-trusted path (with pg12), because in that case I assume that it's possible that there could be two separate users with superuser, so the owner doesn't match and the check fails. @cevian WDYT?

Reply:
in PG12 I think its fine requiring a consistent owning user to run this every time.